src/Symfony/Component/Translation/Translator.php

```diff
@@ -241,36 +241,52 @@ public function transChoice($id, $number, array $parameters = array(), $domain =
         }
 
         $id = (string) $id;
-        $catalogue = $this->getCatalogue($locale);
+        $bestLocale = $this->resolveLocale($id, $domain, $locale);
+
+        if (null !== $bestLocale) {
+           return strtr($this->selector->choose($this->getCatalogueInternal($locale)->get($id, $domain), (int) $number, $bestLocale), $parameters);
```

Comment:
No, that might trigger loading another chain of catalogues (with the one for `bestLocale` at the root). As long as we're still using the fallback chain, it should be fine like this.